Source/monster.cpp

```diff
@@ -3691,6 +3650,75 @@ void GetLevelMTypes()
 	}
 }
 
+void InitMonsterType(int monst)
+{
+	int mtype = LevelMonsterTypes[monst].mtype;
+	LevelMonsterTypes[monst].MData = &MonstersData[mtype];
+	LevelMonsterTypes[monst].mMinHP = MonstersData[mtype].mMinHP;
+	LevelMonsterTypes[monst].mMaxHP = MonstersData[mtype].mMaxHP;
+	if (!gbIsHellfire && mtype == MT_DIABLO) {
+		LevelMonsterTypes[monst].mMinHP /= 2;
+		LevelMonsterTypes[monst].mMaxHP /= 2;
+		LevelMonsterTypes[monst].mLevel -= 15;
+	}
+	
+	if (!gbIsMultiplayer)
+		LevelMonsterTypes[monst].mMinHP = std::max(LevelMonsterTypes[monst].mMinHP / 2, 1);
+		LevelMonsterTypes[monst].mMinHP = std::max(LevelMonsterTypes[monst].mMaxHP / 2, 1);

```

Comment:
This looks wrong. You'll also need to fix a bunch of erroneous indentation that seems to have been caused by this.

```suggestion
	if (!gbIsMultiplayer) {
		LevelMonsterTypes[monst].mMinHP = std::max(LevelMonsterTypes[monst].mMinHP / 2, 1);
		LevelMonsterTypes[monst].mMinHP = std::max(LevelMonsterTypes[monst].mMaxHP / 2, 1);
	}
```